From 7c8dde8eab1a96a0b45f8eaf23a4e991019d3885 Mon Sep 17 00:00:00 2001
From: Bhaskar Upadhaya <Bhaskar.Upadhaya@nxp.com>
Date: Wed, 29 Nov 2017 12:21:43 +0530
Subject: [PATCH 327/741] staging: fsl_ppfe/eth: Disable autonegotiation for
 2.5G SGMII

commit b5444a659e0c68e56f1cf5c16356fc2658a6cdcc from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

PCS initialization sequence for 2.5G SGMII interface governs
auto negotiation to be in disabled mode

Signed-off-by: Bhaskar Upadhaya <Bhaskar.Upadhaya@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/staging/fsl_ppfe/pfe_eth.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_ppfe/pfe_eth.c b/drivers/staging/fsl_ppfe/pfe_eth.c
index 84396ecf3d0d..5c4d7d33e367 100644
--- a/drivers/staging/fsl_ppfe/pfe_eth.c
+++ b/drivers/staging/fsl_ppfe/pfe_eth.c
@@ -1174,6 +1174,7 @@ static void ls1012a_configure_serdes(struct net_device *ndev)
 	struct pfe_eth_priv_s *priv = pfe->eth.eth_priv[0];
 	int sgmii_2500 = 0;
 	struct mii_bus *bus = priv->mii_bus;
+	u16 value = 0;
 
 	if (priv->einfo->mii_config == PHY_INTERFACE_MODE_2500SGMII)
 		sgmii_2500 = 1;
@@ -1191,14 +1192,16 @@ static void ls1012a_configure_serdes(struct net_device *ndev)
 		pfe_eth_mdio_write(bus, 0, 0x4, 0x4001);
 		pfe_eth_mdio_write(bus, 0, 0x12, 0xa120);
 		pfe_eth_mdio_write(bus, 0, 0x13, 0x7);
+		/* Autonegotiation need to be disabled for 2.5G SGMII mode*/
+		value = 0x0140;
+		pfe_eth_mdio_write(bus, 0, 0x0, value);
 	} else {
 		pfe_eth_mdio_write(bus, 0, 0x14, 0xb);
 		pfe_eth_mdio_write(bus, 0, 0x4, 0x1a1);
 		pfe_eth_mdio_write(bus, 0, 0x12, 0x400);
 		pfe_eth_mdio_write(bus, 0, 0x13, 0x0);
+		pfe_eth_mdio_write(bus, 0, 0x0, 0x1140);
 	}
-
-	pfe_eth_mdio_write(bus, 0, 0x0, 0x1140);
 }
 
 /*
-- 
2.17.1

